backend/certificates.js

```diff
@@ -34,3 +34,28 @@ export function saveCertsLocally(json) {
     saveSetting('CONNECTOR_HOST_INFO', hostInfo);
     return {};
 }
+export const msInOneDay = 1000 * 3600 * 24;
+
+export function calculateDaysToRenewal() {
+    const daysCertificateIsGoodFor = 24; // Because max setTimout value is 24 days.
+    const {lastUpdated} = getSetting('CONNECTOR_HOST_INFO');
+    const timePassed = Math.abs(new Date().getTime() - new Date(lastUpdated).getTime());
+    return Math.ceil(daysCertificateIsGoodFor - timePassed / msInOneDay);
+}
+
+export function toMilliseconds(days) {
+    return days * msInOneDay;
+}
+
+export const renewCertificate = () => {
+    // TODO: Not sure what CA endpoint to use at the moment for a renewal.
+    return console.log(`Renewal of ${getSetting('CONNECTOR_HOST_INFO').host}`);
+};
+
+export function setRenewalJob() {
+    // setInterval is using a 32 bit int to store the delay so the max value allowed
+    // would be 2147483647 => 24.8 days.
+    return setInterval(() => {
+        renewCertificate();
+    }, toMilliseconds(calculateDaysToRenewal()));
```

Comment:
Instead of `setInterval`, should this be `setTimeout` and then create a new `setTimeout` on a successful `renewCertificate`? 

For example, if the user opens the app with 2 days left until renewal. Then, they keep the app open for the rest of time. The connector will continuously renew the certificate every 2 days instead of in 2 days on the first round and then 24 days from then on out.